ql/src/java/org/apache/hadoop/hive/ql/io/parquet/vector/VectorizedListColumnReader.java

```diff
@@ -323,7 +341,11 @@ private void fillColumnVector(PrimitiveObjectInspector.PrimitiveCategory categor
       int scale = logicalType.getScale();
       lcv.child = new DecimalColumnVector(total, precision, scale);
       for (int i = 0; i < valueList.size(); i++) {
-        ((DecimalColumnVector) lcv.child).vector[i].set(((List<byte[]>) valueList).get(i), scale);
+        if (valueList.get(i) == null) {
+          lcv.child.isNull[i] = true;
```

Comment:
now that value is not set to null ..compareDecimalColumnVector has to be modified to check fo isNull ..same for other types also

Reply:
right, also compareDecimalColumnVector lacks some inner parentheses here I guess :) I'll fix it
```
        if (cv1.vector[i] != null && cv2.vector[i] == null
            || cv1.vector[i] == null && cv2.vector[i] != null
            || cv1.vector[i] != null && cv2.vector[i] != null && !cv1.vector[i].equals(cv2.vector[i])) {
```
needs to be ((... && ...) || (... && ...) || (... && ... && ...))...without parentheses it is a bit messy/undefined for me...